### ISSUE TITLE
[RFC] Integration Tests

### ISSUE THREAD

**Issue by javiereguiluz** [symfony/symfony-installer#161]
The problem
-----------

Recently we started recommending to use HTTPS for the installer. Sadly, the server infrastructure is not completely ready and this doesn't work. To make things worse, I didn't test the HTTPS feature properly before recommending it.

The solution
------------

The Symfony Installer is an essential tool for Symfony project and it must always work as expected. That's why, besides unit tests, we need **integration tests** to run them for each commit/pull request submitted to this repository.

Specifically, we need to do the following for any new code added to this project:

  1. Build the PHAR file with Box project.
  2. Install Symfony using HTTP and any of the available options (branch, specific version, `lts` special version, etc.)
  3. Install Symfony using HTTPS and any of the available options (branch, specific version, `lts` special version, etc.)
  4. Verify all the previous installations executing at least `php app/console --version` command in the installed Symfony application.

**How can we implement these integration tests?**

**Comment by jakzal** [comment]
What would help in this situation is a smoke test. This could even be a simple bash script that builds the phar and verifies it can be executed with **few** variations. 

We should make it clear that only communication with the outside world should be tested here. This kind of tests are slow and we should have as little of them as possible.

**Comment by desarrolla2** [comment]
I think that you can use symfony/proccess + phpunit to do this.

```php
<?php
use Symfony\Component\Process\Process;

class InstallerTest extends \PHPUnit_Framework_TestCase
{
    public function setUp()
    {
        // set permissions
    }

    public function tearDown()
    {
        // remove files
    }

    public function testBuildPhar()
    {
        $process = new Process('build-phar-file-here');
        $process->run();
        $this->assertTrue($process->isSuccessful());
        
        // verify checksum ? 
    }

    /**
     * @dataProvider branchesDataProvider
     *
     * @param string $branch
     */
    public function testNewProjectWithBranch($branch)
    {
        $process = new Process(sprintf('symfony new my_project %s', $branch));
        $process->run();
        $this->assertTrue($process->isSuccessful());

        // other verifications
    }
}
```

**Comment by Pierstoval** [comment]
Need to ensure it doesn't create too many connections to the `symfony.com` website I guess

**Comment by 94noni** [comment]
Can we not get a master.zip of the installer from github website instead?
In a simple bash script: unzip it, install deps, build phar, and check difference commands for expected outputs (and version/branch/self-update)

**Comment by stof** [comment]
@94noni I don't understand your comment

**Comment by javiereguiluz** [comment]
Implemented in #183. Thank you all for the discussion.
